From 35369b9ef7c3cc87a57b7ac2a5ee3eafc567228f Mon Sep 17 00:00:00 2001
From: Lennart Poettering <lennart@poettering.net>
Date: Tue, 3 Apr 2012 19:25:29 +0200
Subject: [PATCH] logind: log with AUTH facility (cherry picked from commit
 3eff4208ffecedd778fec260f0d4b18e94dab443)

Conflicts:

	TODO
	src/journal/journald.c
	src/log.c
---
 src/log.c    |    7 ++++++-
 src/log.h    |    1 +
 src/logind.c |    1 +
 3 files changed, 8 insertions(+), 1 deletions(-)

diff --git a/src/log.c b/src/log.c
index c7982a2..910daa0 100644
--- a/src/log.c
+++ b/src/log.c
@@ -36,6 +36,7 @@
 
 static LogTarget log_target = LOG_TARGET_CONSOLE;
 static int log_max_level = LOG_INFO;
+static int log_facility = LOG_DAEMON;
 
 static int console_fd = STDERR_FILENO;
 static int syslog_fd = -1;
@@ -252,6 +253,10 @@ void log_set_max_level(int level) {
         log_max_level = level;
 }
 
+void log_set_facility(int facility) {
+        log_facility = facility;
+}
+
 static int write_to_console(
                 int level,
                 const char*file,
@@ -397,7 +402,7 @@ static int log_dispatch(
 
         /* Patch in LOG_DAEMON facility if necessary */
         if ((level & LOG_FACMASK) == 0)
-                level = LOG_DAEMON | LOG_PRI(level);
+                level = log_facility | LOG_PRI(level);
 
         do {
                 char *e;
diff --git a/src/log.h b/src/log.h
index 9942e3e..0a5ceee 100644
--- a/src/log.h
+++ b/src/log.h
@@ -42,6 +42,7 @@ typedef enum LogTarget{
 
 void log_set_target(LogTarget target);
 void log_set_max_level(int level);
+void log_set_facility(int facility);
 
 int log_set_target_from_string(const char *e);
 int log_set_max_level_from_string(const char *e);
diff --git a/src/logind.c b/src/logind.c
index 5cff2db..4cf7586 100644
--- a/src/logind.c
+++ b/src/logind.c
@@ -1199,6 +1199,7 @@ int main(int argc, char *argv[]) {
         int r;
 
         log_set_target(LOG_TARGET_AUTO);
+        log_set_facility(LOG_AUTH);
         log_parse_environment();
         log_open();
 
